src/Libraries/Microsoft.Extensions.Telemetry/Logging/ExtendedLogger.cs

```diff
@@ -0,0 +1,396 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// OPEN TOPICS
+//
+// * When logging stack traces, we include the exception message string. Is this safe from a compliance standpoint?
+//   Should we use the exception summarizer instead?
+//
+// * When using the structured logging extension methods, the formatter currently creates a string of all the
+//   dimension values. Is this what we want? Should we instead improve the console logger to understand dimensions
```

Comment:
> When using the structured logging extension methods

Who is expected to use these? I was thinking these were helpers called by the source generator and not intended for developers to invoke directly.

What creates the formatter in the source generator case when the user doesn't specify a message format in their LoggerMessage attribute, or they explicitly specify an empty string?

Reply:
In the source generator right now, if the customer doesn't supply a log message then the formatter function return an empty string.

I'm going to remove these functions I think, they don't add enough value and I want folks to use the code generator anyway.

Reply:
That sounds fine to me as source generator behavior, its compatible with existing behavior.

What is it that we expect calls the structured logging methods?